src/compiler/scala/tools/nsc/typechecker/Contexts.scala

```diff
@@ -269,6 +269,8 @@ trait Contexts { self: Analyzer =>
     /** A root import is never unused and always bumps context depth. (e.g scala._ / Predef._ and magic REPL imports) */
     def isRootImport: Boolean = false
 
+    var pendingStabilizers: List[Tree] = Nil
```

Comment:
Not ideal to spend a field in every `Context` on what is a niche feature, but I don't have another suggestion right now on how to bubble this data back up after typechecking.

Reply:
Could be a `Map[Context, List[Tree]]` in `CompilationUnit` maybe...

Reply:
I'm okay to defer this to a later performance optimization round.  It seems somewhat similar to the bookeeping in `Macros.{undetparams,delayed,forced},` which is done with sets and maps external to `Context`.